packages/theming/src/elements/ThemeProvider.tsx

```diff
@@ -5,31 +5,72 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React, { PropsWithChildren, useRef } from 'react';
+import React, { PropsWithChildren, useMemo, useRef } from 'react';
 import { ThemeProvider as StyledThemeProvider } from 'styled-components';
 import { useFocusVisible } from '@zendeskgarden/container-focusvisible';
 import { getControlledValue } from '@zendeskgarden/container-utilities';
-import { IThemeProviderProps } from '../types';
+import { IGardenTheme, ISemanticVariables, IThemeProviderProps } from '../types';
 import DEFAULT_THEME from './theme';
+import { getButtonVariables } from './theme/getters';
 import { useDocument } from '../utils/useDocument';
+import { GlobalStyles } from './theme/global-styles';
+
+/**
+ * Builds a semantic theme object and attaches it to a provided (primitive) theme.
+ * If the consumer provided their own variables getter func, use that instead.
+ *
+ * @param {IGardenTheme} theme
+ * @param {string} prefix
+ * @param {ISemanticVariables | undefined} variables
+ * @returns {IGardenTheme}
+ */
+const withVariables = (
+  theme: IGardenTheme,
+  prefix?: string,
+  _variables?: (theme: IGardenTheme) => ISemanticVariables
+): IGardenTheme => {
+  let variables;
+
+  if (_variables) {
+    variables = _variables(theme);
+  } else {
+    variables = {
+      buttons: getButtonVariables(theme)
+    };
```

Comment:
This is an example of the expected return value should a consumer provide their own variables getter function to ThemeProvider.